doc/api/webstreams.md

```diff
@@ -0,0 +1,1096 @@
+# Web Streams API
+
+> Stability: 1 - Experimental
+
+An implementation of the [WHATWG Streams Standard][].
+
+```mjs
+import {
+  ReadableStream,
+  WritableStream,
+  TransformStream,
+} from 'node:stream/web';
+```
+
+```cjs
+const {
+  ReadableStream,
+  WritableStream,
+  TransformStream,
+} = require('stream/web');
+```
+
+## Overview
+
+The [WHATWG Streams Standard][] (or "web streams") defines an API for handling
+streaming data. It is similar to the Node.js [Streams][] API but emerged later
+and has become the "standard" API for streaming data across many JavaScript
+environments.
+
+There are three primary types of objects
+
+* `ReadableStream` - Represents a source of streaming data.
+* `WritableStream` - Represents a destination for streaming data.
+* `TransformStream` - Represents an algorithm for transforming streaming data.
+
+### Example `ReadableStream`
+
+This example creates a simple `ReadableStream` that pushes the current
+`performance.now()` timestamp once every second forever. An async iterable
+is used to read the data from the stream.
+
+```mjs
+import {
+  ReadableStream
+} from 'node:stream/web';
+
+import {
+  setInterval as every
+} from 'node:timers/promises';
+
+import {
+  performance
+} from 'node:perf_hooks';
+
+const SECOND = 1000;
+
+const stream = new ReadableStream({
+  async start(controller) {
+    for await (const _ of every(SECOND))
+      controller.enqueue(performance.now());
+  }
+});
+
+for await (const value of stream)
+  console.log(value);
+```
+
+```cjs
+const {
+  ReadableStream
+} = require('stream/web');
+
+const {
+  setInterval: every
+} = require('timers/promises');
+
+const {
+  performance
+} = require('perf_hooks');
+
+const SECOND = 1000;
+
+const stream = new ReadableStream({
+  async start(controller) {
+    for await (const _ of every(SECOND))
+      controller.enqueue(performance.now());
+  }
+});
```

Comment:
Interesting! I was going to comment that this stream would run forever even when canceled, because it doesn't have a `cancel()` method that stops the `for await of` loop. But actually, this still works: after the stream is canceled, any subsequent calls to `controller.enqueue()` will error, so the next iteration will exit the loop and close the `every()` async iterator. 😅

Reply:
Yeah I had to test it to make sure also and it works really well!